code/datums/elements/decal.dm

```diff
@@ -61,6 +63,15 @@
 
 	if(old_dir == new_dir)
 		return
+
+	var/rotation = 0
+	if(pic.dir != old_dir) //Even when the dirs are the same rotation is coming out as not 0 for some reason
```

Comment:
You'll need to keep track if _dir was set (or check for registered signal on the parent) or decals with hard set dir = 2 won't rotate properly.

Reply:
I don't clearly understand what you mean.
Decals with hard set dir must be unrotateable, mostly it don't have other dirs.
And it don't register rotate_react, and don't rotate.

About passing pic.dir in shuttle_move_react.
Yes it is. And this can break unrotateable decals.
Has idea about make 2 proc, one pass with dir, another with no dir. And one register under `if(_dir)` another after `else`